python/sdss_access/path/path.py

```diff
@@ -1293,5 +1297,52 @@ def configgrp(self, filetype, **kwargs):
             return '0000XX'
         return '{:0>4d}XX'.format(int(configid) // 100)
 
+    def isplate(self, filetype, **kwargs):
+        ''' Returns the plate flag for BOSS idlspec2d run2d versions that utilize it
+
+        Parameters
+        ---------
+        filetype : str
+            File type paramter
+        run2d : str
+            BOSS idlspec2d run2d version
+        
+        Returns
+        -------
+        isplate : str
+            isplate flag = 'p' for relevent run2d plates else flag = ''
+        '''
+
+        run2d = kwargs.get('run2d', None)
+        if not run2d: return ''
+        elif run2d in ['v6_0_1','v6_0_2', 'v6_0_3', 'v6_0_4']: return 'p'
+        else: return ''
+
+    def pad_fieldid(self, filetype, **kwargs):
+        ''' Returns the fieldid zero padded to its proper length for the BOSS idlspec2d run2d version
+
+        Parameters
+        ---------
+        filetype : str
+            File type paramter
+        run2d : str
+            BOSS idlspec2d run2d version
+        fieldid : str or int
+            Field ID number. Will be converted to str internally.
+
+        Returns
+        -------
+        padd_fieldid : str
+            padd_fieldid in the form of N*'0' where N is the number of necessary zeros to pad fieldid
+        '''
+        run2d = kwargs.get('run2d', None)
+        if not run2d: return ''
+        elif run2d in ['v6_0_1','v6_0_2', 'v6_0_3', 'v6_0_4']: return ''
+        else: 
+            fieldid = kwargs.get('fieldid', None)
+            if not fieldid: return ''
+            if len(str(fieldid)) < 6: return '0'*(6 - len(str(fieldid)))
+            else: return ''
```

Comment:
Since you're returning in each conditional block, you can avoid unnecessary nesting and conditionals by moving everything up one level or so.  
```suggestion
        if not run2d:
            return ''

        if run2d in ['v6_0_1','v6_0_2', 'v6_0_3', 'v6_0_4']:
            return ''

        fieldid = kwargs.get('fieldid', None)
        if not fieldid:
            return ''
           
        return '0'*(6 - len(str(fieldid))) if len(str(fieldid)) < 6 else ''
```